File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int curr_max = 0, max_end = Integer.MIN_VALUE;
        for(int i = 0; i < nums.length; i++) {
            curr_max += nums[i];
            if(curr_max > max_end) {
                max_end = curr_max;
            }
            if(curr_max < 0) {
                curr_max = 0;
            }
        }
        return max_end;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int maxans = INT_MIN;
        int sum = 0;
        for(int i = 0; i < n; i++) {
            sum += nums[i];
            maxans = max(maxans,  sum);
            if(sum < 0) {
                sum = 0;
            }
        }
        return maxans;
    }
};